x-pack/legacy/plugins/alerting_ui/np_ready/public/application/sections/actions_list/components/actions_list.tsx

```diff
@@ -0,0 +1,264 @@
+
+  async function loadActionsTable() {
+    setIsLoadingActions(true);
+    try {
+      const actionsResponse = await loadActions({ http, page, searchText });
+      setActions(actionsResponse.data);
+      setTotalItemCount(actionsResponse.total);
+    } catch (e) {
+      toastNotifications.addDanger({
+        title: i18n.translate('xpack.alertingUI.sections.actionsList.unableToLoadActionsMessage', {
+          defaultMessage: 'Unable to load actions',
+        }),
+      });
+    } finally {
+      setIsLoadingActions(false);
+    }
+  }
+
+  async function deleteItems(items: Data[]) {
+    setIsDeletingActions(true);
+    await deleteActions({ http, ids: items.map(item => item.id) });
+    await loadActionsTable();
+    setIsDeletingActions(false);
+  }
+
+  async function deleteSelectedItems() {
+    await deleteItems(selectedItems);
+  }
+
+  const actionsTableColumns = [
+    {
+      field: 'description',
+      name: i18n.translate(
+        'xpack.alertingUI.sections.actionsList.actionsListTable.columns.description',
+        {
+          defaultMessage: 'Description',
+        }
+      ),
+      sortable: false,
+      truncateText: true,
+    },
+    {
+      field: 'actionType',
+      name: i18n.translate(
+        'xpack.alertingUI.sections.actionsList.actionsListTable.columns.actionType',
+        {
+          defaultMessage: 'Action Type',
+        }
+      ),
+      sortable: false,
+      truncateText: true,
+    },
+    {
+      field: 'config',
+      name: i18n.translate(
+        'xpack.alertingUI.sections.actionsList.actionsListTable.columns.config',
+        { defaultMessage: 'Config' }
+      ),
+      sortable: false,
+      truncateText: false,
+    },
+    {
+      name: i18n.translate(
+        'xpack.alertingUI.sections.actionsList.actionsListTable.columns.actions',
+        { defaultMessage: 'Actions' }
+      ),
+      actions: [
+        {
+          enabled: () => canDelete,
+          name: i18n.translate(
+            'xpack.alertingUI.sections.actionsList.actionsListTable.columns.actions.deleteActionName',
+            { defaultMessage: 'Delete' }
+          ),
+          description: canDelete
+            ? i18n.translate(
+                'xpack.alertingUI.sections.actionsList.actionsListTable.columns.actions.deleteActionDescription',
+                { defaultMessage: 'Delete this action' }
+              )
+            : i18n.translate(
+                'xpack.alertingUI.sections.actionsList.actionsListTable.columns.actions.deleteActionDisabledDescription',
+                { defaultMessage: 'Unable to delete actions' }
+              ),
+          type: 'icon',
+          icon: 'trash',
+          onClick: (item: Data) => deleteItems([item]),
+        },
+      ],
+    },
+  ];
+
+  return (
+    <section data-test-subj="actionsList">
+      <ContentWrapper>
+        <EuiSpacer size="m" />
+        <EuiSearchBar
+          onChange={({ queryText }: { queryText: string }) => setSearchText(queryText)}
```

Comment:
we might want to debounce this.  as it sits, we will be querying ES with every keystroke.

Reply:
As discussed, now that we're using in memory table, no longer an issue. Fixed in https://github.com/elastic/kibana/pull/47843/commits/afffcf36905e7b34ebfa5188a367e19a5896541d.